sematic/ui/packages/main/src/types/prompt_response.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { SyntheticEvent, useState } from "react";
+import Accordion from "@mui/material/Accordion";
+import AccordionSummary from "@mui/material/AccordionSummary";
+import AccordionDetails from "@mui/material/AccordionDetails";
+import ExpandMoreIcon from "@mui/icons-material/ExpandMore";
+import Typography from "@mui/material/Typography";
+import { CommonValueViewProps } from "./common";
+
+const SUMMARY_MAX_LENGTH = 100;
+
+export default function PromptResponseValueView(props: CommonValueViewProps) {
+    let { valueSummary } = props;
+    let { values } = valueSummary;
+    const prompt = values.prompt;
+    const response = values.response;
+
+    const promptComponent = TextSummaryAccordion({text: prompt, textKind: "Prompt"});
+    const responseComponent = TextSummaryAccordion({text: response, textKind: "Response"});
+
+    return (
+        <div>
+            {promptComponent}
+            {responseComponent}
+        </div>
+    );
+}
+
+interface TextSummaryAccordionProps {
+    text: string;
+    textKind: string;
+}
+
+function TextSummaryAccordion(props: TextSummaryAccordionProps) {
+    let { text, textKind } = props;
+    const [expanded, setExpanded] = useState<boolean>(false);
+
+    const handleChange = () => (event: SyntheticEvent, isExpanded: boolean) => {
+        setExpanded(!expanded);
+    };
+    const textIsLong = text.length > SUMMARY_MAX_LENGTH; 
+    const shortText = (
+        textIsLong ?
+            text.substring(0, SUMMARY_MAX_LENGTH) + "..." :
+            text
+    );
+    const textJsx = text.split("\n").map((str: string) => <p>{str}</p>);
```

Comment:
Another way we adopt pervasively in this code base to deal with such cases is to use `<pre>all_text</pre>` to preserve paragraph format instead of creating multiple `<P>`S

Reply:
That formats the string in monospace, and it seems to allow overflows (perhaps because I'm using the Typography element?). So I'll leave as-is.

Reply:
yeah, additional styling will be needed on the `<pre>` though. I am just saying it saves multiple HTML tags.

Leaving as-is is perfectly fine too.